File: client/java/src/avt/StarFruitObj.java
package avt;

import javax.microedition.lcdui.Graphics;
import main.Canvas;

public final class StarFruitObj extends SubObject {
   public short lv;
   public short productID;
   public short fruitID;
   public short numberFruit;
   public int timeFinish;
   private int w0 = 0;
   private int h0 = 0;
   public long time;
   public byte[] xFruit;
   public byte[] yFruit;

   public final void update() {
      if (System.currentTimeMillis() - this.time >= 1000L) {
         if (this.timeFinish > 0) {
            --this.timeFinish;
            if (this.timeFinish == 0) {
               FarmService var1;
               (var1 = FarmService.gI()).createMessage((byte)83);
               var1.sendMessage();
            }
         }

         this.time = System.currentTimeMillis();
         ImageIcon var4;
         if ((var4 = FarmData.getImgIcon(this.productID)).w > 0 && this.w0 == 0) {
            this.w0 = var4.w / 3 << 1;
            this.h0 = var4.h / 2;
            StarFruitObj var5 = this;
            if (this.numberFruit > 0) {
               int var2 = CRes.rnd(3) + 3;
               this.xFruit = new byte[var2];
               this.yFruit = new byte[var2];

               for(int var3 = 0; var3 < var2; ++var3) {
                  var5.xFruit[var3] = (byte)(CRes.rnd(var5.w0 - 10) - (var5.w0 - 10) / 2);
                  var5.yFruit[var3] = (byte)(CRes.rnd(var5.h0 - 10) - (var5.h0 - 10) / 2);
               }
            }
         }
      }

   }

   public final void paint(Graphics var1) {
      if (super.type >= 0 || super.x * MyObject.hd + this.w0 / 2 >= AvCamera.gI().xCam && super.x * MyObject.hd - this.w0 / 2 <= AvCamera.gI().xCam + Canvas.w) {
         FarmData.paintImg(var1, this.productID, super.x * MyObject.hd, super.y * MyObject.hd, 33);
         int var2;
         if (this.numberFruit > 0 && this.xFruit != null) {
            for(var2 = 0; var2 < this.xFruit.length; ++var2) {
               FarmData.paintImg(var1, this.fruitID, super.x * MyObject.hd + this.xFruit[var2], super.y * MyObject.hd - (FarmData.getImgIcon(this.productID).h / 2 + 5) + this.yFruit[var2], 3);
            }
         }

         var2 = FarmData.getImgIcon(this.productID).h + AvMain.hBorder;
         if (this.timeFinish > 0) {
            var2 += AvMain.hSmall;
         }

         FarmData.paintImg(var1, this.fruitID, (super.x - 8) * MyObject.hd, super.y * MyObject.hd - var2, 3);
         Canvas.borderFont.drawString(var1, "Lv" + this.lv, super.x * MyObject.hd, super.y * MyObject.hd - var2 - AvMain.hBorder / 2, 0);
         if (this.timeFinish > 0) {
            int var3 = this.timeFinish / 3600;
            int var4 = (this.timeFinish - var3 * 3600) / 60;
            int var5 = this.timeFinish - var3 * 3600 - var4 * 60;
            Canvas.smallFontYellow.drawString(var1, var3 + ":" + var4 + ":" + var5, (super.x + 3) * MyObject.hd, super.y * MyObject.hd - var2 + Canvas.borderFont.getHeight() / 2 + 2 * MyObject.hd, 2);
         }
      }

   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/StarFruitObj.cs
public class StarFruitObj : SubObject
{
	public short lv;

	public short productID;

	public short fruitID;

	public short numberFruit;

	public short anTrom;

	public static int imgID;

	public int timeFinish;

	public int w0;

	public int h0;

	public long time;

	public sbyte[] xFruit;

	public sbyte[] yFruit;

	public StarFruitObj()
	{
		catagory = 8;
	}

	public void setFruit()
	{
		if (numberFruit > 0)
		{
			int num = CRes.rnd(3) + 3;
			xFruit = new sbyte[num];
			yFruit = new sbyte[num];
			for (int i = 0; i < num; i++)
			{
				xFruit[i] = (sbyte)(CRes.rnd(w0 - 10) - (w0 - 10) / 2);
				yFruit[i] = (sbyte)(CRes.rnd(h0 - 10) - (h0 - 10) / 2);
			}
		}
	}

	public override void update()
	{
		type = imgID;
		if (Canvas.getTick() - time < 1000)
		{
			return;
		}
		if (timeFinish > 0)
		{
			timeFinish--;
			if (timeFinish == 0)
			{
				FarmService.gI().doFinishStarFruit();
			}
		}
		time = Canvas.getTick();
		ImageIcon imgIcon = FarmData.getImgIcon((short)imgID);
		if (imgIcon.w > 0)
		{
			if (w == 0)
			{
				w = imgIcon.w;
				h = imgIcon.h;
			}
			if (w0 == 0)
			{
				w0 = imgIcon.w / 3 * 2;
				h0 = imgIcon.h / 2;
				setFruit();
			}
		}
	}

	public override void paint(MyGraphics g)
	{
		if (type < 0 && ((float)(x * MyObject.hd + w / 2) < AvCamera.gI().xCam || (float)(x * MyObject.hd - w / 2) > AvCamera.gI().xCam + (float)Canvas.w))
		{
			return;
		}
		FarmData.paintImg(g, imgID, x * MyObject.hd, y * MyObject.hd, MyGraphics.HCENTER | MyGraphics.BOTTOM);
		if (numberFruit > 0 && xFruit != null)
		{
			for (int i = 0; i < xFruit.Length; i++)
			{
				FarmData.paintImg(g, fruitID, x * MyObject.hd + xFruit[i], y * MyObject.hd - (FarmData.getImgIcon((short)imgID).h / 2 + 5) + yFruit[i], 3);
			}
		}
		int num = 0;
		num = FarmData.getImgIcon((short)imgID).h + AvMain.hBorder;
		if (timeFinish > 0)
		{
			num += AvMain.hSmall;
		}
		FarmData.paintImg(g, fruitID, (x - 8) * MyObject.hd, y * MyObject.hd - num, 3);
		Canvas.borderFont.drawString(g, "Lv" + lv, x * MyObject.hd, y * MyObject.hd - num - AvMain.hBorder / 2, 0);
		if (timeFinish > 0)
		{
			int num2 = timeFinish / 3600;
			int num3 = (timeFinish - num2 * 3600) / 60;
			int num4 = timeFinish - num2 * 3600 - num3 * 60;
			Canvas.smallFontYellow.drawString(g, num2 + ":" + num3 + ":" + num4, (x + 3) * MyObject.hd, y * MyObject.hd - num + Canvas.borderFont.getHeight() / 2 + 2 * MyObject.hd, 2);
		}
	}
}
